dynawo/sources/Models/Modelica/Dynawo/Electrical/Switches/IdealSwitch.mo

```diff
@@ -0,0 +1,44 @@
+within Dynawo.Electrical.Switches;
+
+/*
+* Copyright (c) 2015-2019, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source time domain simulation tool for power systems.
+*/
+
+model IdealSwitch "Ideal switch"
+  /*
+    When the switch is closed, the voltage on both terminals are equal and the current is going through the switch.
+    When the switch is open, the current going through the switch is equal to zero.
+
+    Equivalent circuit and conventions:
+
+               I1                  I2
+    (terminal1) -->-------/ -------<-- (terminal2)
+
+  */
+
+  import Dynawo.Connectors;
+
+  Connectors.ACPower terminal1 "Switch side 1";
+  Connectors.ACPower terminal2 "Switch side 2";
+  Connectors.BPin closed (value(start = true)) "True when the switch is closed, false otherwise";
```

Comment:
Should be use the state enum (open, closed, closed1, closed2) and have something in Controls>Basics>SwitchOff as for lines and transformers?

Reply:
Done